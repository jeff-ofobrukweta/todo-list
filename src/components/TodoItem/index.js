import React from 'react'
import PropTypes from 'prop-types'
import {
  Container,
  Task,
  RemoveButton,
  Content,
  StyledLink,
  CompleteTask,
  IncompleteTask,
  FlexContainerWrapper,
  CompleteTaskText,
} from './styles'

const TodoItem = (props) => {
  const { className, onCheckboxChange, task, id, onRemoveClick, completed } =
    props

  return (
    <Container className={className}>
      <RemoveButton onClick={onRemoveClick}>x</RemoveButton>
      <Content>
        {completed ? (
          <CompleteTaskText>{task}</CompleteTaskText>
        ) : (
          <Task title={task}>{task}</Task>
        )}

        <FlexContainerWrapper>
          <StyledLink to={`/${id}`}>Edit</StyledLink>
          {completed ? (
            <CompleteTask onClick={onCheckboxChange}>
              <i className="fas fa-check"></i>
            </CompleteTask>
          ) : (
            <IncompleteTask onClick={onCheckboxChange}>
              <i className="fas fa-grip-lines"></i>
            </IncompleteTask>
          )}
        </FlexContainerWrapper>
      </Content>
    </Container>
  )
}

TodoItem.propTypes = {
  className: PropTypes.string,
  task: PropTypes.string,
  completed: PropTypes.bool,
  onCheckboxChange: PropTypes.func,
  handleChange: PropTypes.func,
}

export default TodoItem
