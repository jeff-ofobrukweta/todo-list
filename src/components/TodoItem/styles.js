import styled from 'styled-components'
import { Link } from 'react-router-dom'

export const Container = styled.div`
  border-radius: 5px;
  border: 1px solid rgba(255, 255, 255, 0.1);
  padding: 1.6rem;
  position: relative;
  overflow: hidden;
  display: flex;
  align-items: center;
  transition: all 0.2s ease-in-out;

  &:hover {
    border: 1px solid rgba(255, 255, 255, 0.2);
  }
`

export const Content = styled.div`
  flex: 1;
  display: flex;
  align-items: center;
  justify-content: space-between;
  margin-left: 1.6rem;
  margin-right: 2.4rem;
  overflow: hidden;
`

export const Task = styled.div`
    width: 270px;
    white-space: nowrap;
    overflow: hidden;
    text-overflow: ellipsis;
`

export const CompleteTaskText = styled(Task)`
  text-decoration: line-through;
  border: none;
  box-shadow: none;
  color: grey;
`

export const CompleteTask = styled.button`
  color: #fff !important;
  background: var(--accent-color);
  border-radius: 5px;
  padding: 5px 10px;
  margin: 0px 1rem;
  border: none;
  cursor: pointer;
  font-weight: bold;

  background-position: center;
  transition: background 0.8s;

  &:hover {
    background: #ed7c83 radial-gradient(circle, transparent 1%, #ed7c83 1%)
      center/15000%;
  }

  &:active {
    background-color: #e3babc !important;
    background-size: 100% !important;
    transition: background 0s !important;
  }
`

export const IncompleteTask = styled(CompleteTask)`
  background: #8e8e8e2e;
  background-position: center;
  transition: background 0.8s;

  &:hover {
    background: #3a3a3a radial-gradient(circle, transparent 1%, #3a3a3a 1%)
      center/15000%;
  }

  &:active {
    background-color: #444444 !important;
    background-size: 100% !important;
    transition: background 0s !important;
  }
`

export const SingleInCompleteTask = styled(IncompleteTask)`
  padding: 0px 4px;
  margin: 0px 1rem 0rem 0rem;
`
export const SingleCompleteTask = styled(CompleteTask)`
  padding: 0px 4px;
  margin: 0px 1rem 0rem 0rem;
`

export const RemoveButton = styled.div`
  position: absolute;
  top: 0;
  right: 0;
  width: 3rem;
  height: 3rem;
  display: flex;
  justify-content: center;
  align-items: center;
  transition: all 0.2s ease-in-out;
  cursor: pointer;
  font-weight: bold;
  background: rgba(255, 255, 255, 0.1);
  border-bottom-left-radius: 5px;

  &:hover {
    color: var(--accent-color);
  }
`
export const FlexContainerWrapper = styled.div`
  display: flex;
  justify-content: space-between;
  align-item: center;
`

export const StyledLink = styled(Link)`
  text-decoration: none;
  color: #fff;
  padding: 5px 10px;
  border-radius: 5px;
  background: #8e8e8e2e;
`
