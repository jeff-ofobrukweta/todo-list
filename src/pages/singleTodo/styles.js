import styled, { keyframes } from 'styled-components'

export const Container = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  min-height: 100%;
  padding: 3.2rem 0;
`

export const Title = styled.h1`
  color: var(--text-color);
  margin-bottom: 3.2rem;
  text-align: center;
`

export const InputContainer = styled.div`
  display: flex;
  justify-content: space-between;
  align-items: center;
  flex-wrap: wrap;
  width: 100%;
`

export const Input = styled.input`
  width: 100%;
  padding: 1.6rem;
  background: var(--background-color);
  border: none;
  border: 2px solid rgba(255, 255, 255, 0.6);
  border-radius: 5px;
  transition: all 0.2s ease-in-out;
  outline: none;
  margin: 3rem auto;

  &:focus {
    background: rgba(0, 0, 0, 0.1);
    border: 2px solid rgba(255, 255, 255, 0.8);
  }
`

const ripple = keyframes`
0% {
  transform: scale(0, 0);
  opacity: 1;
}
20% {
  transform: scale(25, 25);
  opacity: 1;
}
100% {
  opacity: 0;
  transform: scale(40, 40);
}`

export const AddButton = styled.button`
  padding: 1.6rem 2.4rem;
  background: var(--accent-color);
  transition: background 0.2s ease-in-out;
  outline: none;
  border: none;
  border-radius: 5px;
  margin: 0.8rem 0 3.2rem 0;
  width: 100%;
  cursor: pointer;
  font-weight: bold;
  font-size: 1.8rem;

  background-position: center;
  transition: background 0.8s;

  background-position: center;
  transition: background 0.8s;

  &:hover {
    background: #ed7c83 radial-gradient(circle, transparent 1%, #ed7c83 1%)
      center/15000%;
  }

  &:active {
    background-color: #e3babc !important;
    background-size: 100% !important;
    transition: background 0s !important;
  }
`

export const TodoListContainer = styled.div`
  background: var(--list-container-background);
  padding: 3.2rem;
  border-radius: 5px;
  width: 100%;
`

export const EmptyListMessage = styled.div`
  color: rgba(255, 255, 255, 0.4);
  text-align: center;
`
