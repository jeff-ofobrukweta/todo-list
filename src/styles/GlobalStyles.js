import { createGlobalStyle } from 'styled-components'

export default createGlobalStyle`
 :root{
    --accent-color: #f36b74;;
    --accent-color-active: #ed7c83d4;
    --text-color: #ffffff;
    --background-color: #49486c;
    --list-container-background: #232323;
    --error-color: #ff4d4d;
  }

  * {
    box-sizing: border-box;
    margin: 0;
    padding: 0;
    outline: 0;
  }

  html {
    font-size: 62.5%;
  }

  body, textarea, input, button {
    color: var(--text-color);
    font-size: 1.6rem;
    -webkit-font-smoothing: antialiased;
    -moz-osx-font-smoothing: grayscale;
    font-family: 'Poppins', sans-serif;
  }

  body {
    background: var(--background-color);
  }

  html, body, #root {
    height: 100%;
  }

button:disabled,
button[disabled]{
  cursor: not-allowed;
}
`
