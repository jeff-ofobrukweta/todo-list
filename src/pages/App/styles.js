import styled, { keyframes } from 'styled-components'
import TodoItem from 'components/TodoItem'
import ErrorBox from 'components/ErrorBox'

export const Container = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  min-height: 100%;
  padding: 3.2rem 0;
`
export const DivContainer = styled.div`
  width: 100%;
  display: flex;
  align-items: center;
`

export const TodoContainer = styled.form`
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  width: 40%;
  max-width: 500px;
  padding: 0rem 1rem;
  width: 100%;
`

export const SingleTodoContainer = styled(TodoContainer)`
  transform: translateY(87%);
`

export const Title = styled.h1`
  color: var(--text-color);
  margin-bottom: 3.2rem;
  text-align: center;
`

export const InputContainer = styled.div`
  display: flex;
  justify-content: space-between;
  align-items: center;
  flex-wrap: wrap;
  width: 100%;
`

const rotate = keyframes`
  0% {
    border-bottom-left-radius: 0px;
  }
  50% {
    border-bottom-left-radius: 40px;
  }
  100% {
    border-bottom-left-radius: 0px;
  }
`

export const Input = styled.input`
  margin: 8px 0;
  width: 100%;
  padding: 1.6rem;
  background: var(--background-color);
  border: none;
  border-bottom: 2px solid rgba(255, 255, 255, 0.6);
  ${'' /* border-radius: 5px; */}
  ${'' /* transition: all 0.2s ease-in-out; */}
  ${'' /* outline: none; */}
${
    '' /* 
  @keyframes bump {
    0% {
      border-bottom-left-radius: 0px;
    }
    50% {
      border-bottom-left-radius: 40px;
    }
    100% {
      border-bottom-left-radius: 0px;
    }
  } */
  }

  

  &:focus {
    ${'' /* background: rgba(0, 0, 0, 0.1); */}
    outline : none;
    ${'' /* border: 2px solid rgba(255, 255, 255, 0.8); */}
    animation: ${rotate} .2s;
  }
  ::-webkit-input-placeholder {
    color: rgba(255, 255, 255, 0.8);
    -webkit-transition: ease-in 0.3s;
    transition: ease-in 0.3s;
    -webkit-transform-origin: 0 50%;
    transform-origin: 0 50%;
  }
  :focus::-webkit-input-placeholder {
    opacity: 0;
    transform: translate(70%);
    -webkit-transition: all 0.35s ease-in-out;
    transition: all 0.35s ease-in-out;
  }
`

export const AddButton = styled.button`
  padding: 1.6rem 2.4rem;
  background: var(--accent-color);
  transition: background 0.3s ease-in-out;
  outline: none;
  border: none;
  border-radius: 5px;
  margin: 0.8rem 0 3.2rem 0;
  width: 100%;
  cursor: pointer;
  font-weight: bold;
  font-size: 1.8rem;

  background-position: center;
  transition: background 0.8s;

  &:hover {
    background: #ed7c83 radial-gradient(circle, transparent 1%, #ed7c83 1%)
      center/15000%;
  }

  &:active {
    background-color: #e3babc !important;
    background-size: 100% !important;
    transition: background 0s !important;
  }
`

export const TodoListContainer = styled.div`
  background: var(--list-container-background);
  padding: 1.2rem;
  border-radius: 5px;
  width: 100%;
`

export const EmptyListMessage = styled.div`
  color: rgba(255, 255, 255, 0.4);
  text-align: center;
`

export const TaskItem = styled(TodoItem)`
  margin-bottom: 8px;
`

export const Error = styled(ErrorBox)`
  margin-bottom: 1.6rem;
`
