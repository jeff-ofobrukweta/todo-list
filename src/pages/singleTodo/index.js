import React, { useEffect, useState } from 'react'
import { useDispatch, useSelector } from 'react-redux'
import { Container, DivContainer } from '../App/styles'
import { useParams } from 'react-router'
import { useNavigate } from 'react-router-dom'
import {
  AddButton,
  Input,
  InputContainer,
  SingleTodoContainer,
} from 'pages/App/styles'
import { updateTaskById } from 'store/ducks/task/actions'
import {
  SingleCompleteTask,
  SingleInCompleteTask,
} from '../../components/TodoItem/styles'

const SingleTodo = () => {
  const [params, setParams] = useState(useParams().todo)
  const [task, setTask] = useState('')
  const [todo, setTodo] = useState({})
  const { taskList } = useSelector(({ Task }) => Task)
  const navigate = useNavigate()
  const dispatch = useDispatch()

  const handleSave = (e) => {
    e.preventDefault()
    setTodo({ ...todo, task })
    if (!task.trim()) return
    dispatch(
      updateTaskById({
        payload: { ...todo, task },
      }),
    )
    navigate('/')
  }

  const toggleCompleted = () => {
    setTodo({ ...todo, completed: !todo.completed })
  }

  useEffect(() => {
    if (
      !taskList.length ||
      !taskList.filter((todo) => Number(todo.id) === Number(params)).length
    ) {
      navigate('/')
    }
  }, [taskList])

  useEffect(() => {
    const singleTodo = taskList.filter(
      (todo) => Number(todo.id) === Number(params),
    )
    setTodo(singleTodo[0])
    setTask(singleTodo[0]?.task)
  }, [params])

  return (
    <>
      <Container>
        <SingleTodoContainer onSubmit={handleSave}>
          <InputContainer>
            <Input onChange={(e) => setTask(e.target.value)} value={task} />
          </InputContainer>
          <DivContainer>
            {todo.completed ? (
              <SingleCompleteTask onClick={toggleCompleted} type="button">
                <i className="fas fa-check"></i>
              </SingleCompleteTask>
            ) : (
              <SingleInCompleteTask onClick={toggleCompleted} type="button">
                <i className="fas fa-grip-lines"></i>
              </SingleInCompleteTask>
            )}
            <DivContainer>
              {todo.completed ? 'Completed' : 'Not Completed'}{' '}
            </DivContainer>
          </DivContainer>
          <AddButton
            disabled={!task.trim().length ? 'disabled' : ''}
            type="submit"
          >
            Save Changes
          </AddButton>
        </SingleTodoContainer>
      </Container>
    </>
  )
}

export default SingleTodo
