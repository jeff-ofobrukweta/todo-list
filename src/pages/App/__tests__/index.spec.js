import React from 'react';
import { shallow } from 'enzyme';
import TodoItem from 'components/TodoItem';
import { TodoListContainer } from 'pages/singleTodo/styles';




describe('Title', () => {
  
  it('should snapshot the TodoItem Component correctly', () => {   
    let wrapped = shallow(<TodoItem/>);
    expect(wrapped).toMatchSnapshot();
  });

  it('should snapshot the TodoListContainer Component correctly', () => {   
    let wrapped = shallow(<TodoListContainer/>);
    expect(wrapped).toMatchSnapshot();
  });
});