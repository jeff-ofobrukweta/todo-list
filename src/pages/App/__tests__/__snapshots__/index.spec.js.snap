// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title should snapshot the TodoItem Component correctly 1`] = `
<styled.div>
  <styled.div>
    x
  </styled.div>
  <styled.div>
    <styled.div />
    <styled.div>
      <Styled(Link)
        to="/undefined"
      >
        Edit
      </Styled(Link)>
      <Styled(styled.button)>
        <i
          className="fas fa-grip-lines"
        />
      </Styled(styled.button)>
    </styled.div>
  </styled.div>
</styled.div>
`;

exports[`Title should snapshot the TodoListContainer Component correctly 1`] = `
<div
  className="sc-fzokOt hVigUA"
/>
`;
